src/app/shell/shell.component.ts

```diff
@@ -108,7 +108,11 @@ export class ShellComponent implements OnInit {
 	redirect() {
 		this.cookieService.set('org', this.organization.url, null, '/', '.newvote.org');
 		// window.location.href = this.org.authUrl;
-		window.open(this.organization.authUrl, '_self');
+		if (this.auth.isAdmin()) {
+			return this.router.navigate(['/auth/login?admin="true']);
+		}
+
+		return this.router.navigate(['/auth/login']);
 	}
```

Comment:
Good thinking, however auth.isAdmin only works if the user is already logged in, how will this work?
Just move the routing onto the template, and remove this redirect function.